src/manifolds/SymmetricPositiveDefiniteGeneralizedBuresWasserstein.jl

```diff
@@ -0,0 +1,144 @@
+    Y,
+    ::EuclideanMetric,
+    p,
+    X,
+)
+    Y .= 2 .* (p * X * M.metric.M + M.metric.M * X * p)
+    return Y
+end
+
+@doc raw"""
+    distance(::MatricManifold{SymmetricPositiveDefinite,GeneralizedBuresWassersteinMetric}, p, q)
+
+Compute the distance with respect to the [`BuresWassersteinMetric`](@ref) on [`SymmetricPositiveDefinite`](@ref) matrices, i.e.
+
+```math
+    d(p.q) = \bigl( \operatorname{tr}(M^{-1}p) + \operatorname{tr}(M^{-1}q) + 2\operatorname{tr}(p^{\frac{1}{2}M^{-1}qM^{-1}p^{\frac{1}[2}) \bigr)^\frac{1}[2},
+```
+
+where the last trace can be simplified (by rotating the matrix products in the trace) to ``\operatorname{tr}(M^{-1}pM^{-1}q)``.
+"""
+function distance(
+    M::MetricManifold{ℝ,SymmetricPositiveDefinite,GeneralizedBuresWassersteinMetric},
+    p,
+    q,
+)
+    luM = lu(M.metric.M)
+    return sqrt(tr(luM \ p) + tr(luM \ q) + 2 * tr((lum \ q) * (lum \ p)))
+end
+
+@doc raw"""
+    exp(::MatricManifold{ℝ,SymmetricPositiveDefinite,GeneralizedBuresWassersteinMetric}, p, X)
+
+Compute the exponential map on [`SymmetricPositiveDefinite`](@ref) with respect to
+the [`GeneralizedBuresWassersteinMetric`](@ref) given by
+
+```math
+    \exp_p(X) = p+X+\mathcal ML_{p,M}(X)pML_{p,M}(X)
+```
+
+where ``q=L_{M,p}(X)`` denotes the generalized lyaponov operator, i.e. it solves ``pqM + Mqp = X``.
+"""
+exp(::MetricManifold{ℝ,SymmetricPositiveDefinite,GeneralizedBuresWassersteinMetric}, p, X)
+
+function exp!(
+    M::MetricManifold{ℝ,SymmetricPositiveDefinite,GeneralizedBuresWassersteinMetric},
+    q,
+    p,
+    X,
+)
+    q .= lyapc(p, M.metric.M, -X) #lyap solves qpM + Mpq - X =0
+    q .= p .+ X .+ M.metricM * q * p * q * M.metric.M
+    return Y
+end
+
+@doc raw"""
+    inner(::MetricManifold{ℝ,SymmetricPositiveDefinite,GeneralizedBuresWassersteinMetric}, p, X, Y)
+
+Compute the inner product [`SymmetricPositiveDefinite`](@ref) with respect to
+the [`GeneralizedBuresWassersteinMetric`](@ref) given by
+
+```math
+    ⟨X,Y⟩ = \frac{1}{2}\operatorname{tr}( \mathcal L_{p,M}(X)Y)
+```
+
+where ``q=L_{M,p}(X)`` denotes the generalized lyaponov operator, i.e. it solves ``pqM + Mqp = X``.
+"""
+function inner(
+    M::MetricManifold{ℝ,SymmetricPositiveDefinite,GeneralizedBuresWassersteinMetric},
+    p,
+    X,
+    Y,
+)
+    return 1 / 2 * tr(lyapc(p, M.metric.M, -X) * Y)
+end
+
+@doc raw"""
+    log(::MatricManifold{SymmetricPositiveDefinite,GeneralizedBuresWassersteinMetric}, p, q)
+
+Compute the logarithmic map on [`SymmetricPositiveDefinite`](@ref) with respect to
+the [`BuresWassersteinMetric`](@ref) given by
+
+```math
+    \log_p(q) = M(M^{-1}pM^{-1}q)^{\frac{1}{2}} + (qM^{-1}pM^{-1})^{\frac{1}{2}}M^{-1} - 2 p
+```
+
+where ``q=L_p(X)`` denotes the lyaponov operator, i.e. it solves ``pq + qp = X``.
+"""
+log(::MetricManifold{ℝ,SymmetricPositiveDefinite,GeneralizedBuresWassersteinMetric}, p, q)
+
+function log!(
+    M::MetricManifold{ℝ,SymmetricPositiveDefinite,GeneralizedBuresWassersteinMetric},
```

Comment:
```suggestion
    M::MetricManifold{ℝ,<:SymmetricPositiveDefinite,GeneralizedBuresWassersteinMetric},
```
You need to allow for concrete types there. `BuresWassersteinMetric` has the same issue.

Reply:
Basically all methods here have the same issue.

Reply:
And why does it work on LogCholesky? Because specifying the N makes them concrete?

Reply:
Yes, exactly.

Reply:
Hm, does not seem to resolve the issue.

Reply:
Ah, in this case you also need `<:GeneralizedBuresWassersteinMetric`.

Reply:
Well spotted. Thanks